integration-tests/src/main/java/org/apache/polaris/service/it/test/PolarisRestCatalogViewIntegrationBase.java

```diff
@@ -201,16 +201,10 @@ public void createViewWithCustomMetadataLocationUsingPolaris(@TempDir Path tempD
     TableIdentifier identifier = TableIdentifier.of("ns", "view");
 
     String location = Paths.get(tempDir.toUri().toString()).toString();
```

Comment:
This location still escape from the storage config. We will need more fixes.

Reply:
The base class does add the `@TempDir` to allowedLocations in https://github.com/apache/polaris/blob/f555f30f6f6a10df441e8da84098c82245de28a9/integration-tests/src/main/java/org/apache/polaris/service/it/test/PolarisRestCatalogViewFileIntegrationTestBase.java#L43

Is this a different tempdir that escapes? If we want to exercise a failure we probably need to try to set the `location` to something else.

Reply:
Weird. It works as expected when I moved the integration test to unit test.

Reply:
It turns out the Polaris added a structured location to the view in https://github.com/apache/polaris/pull/1966. When a rest client creates view with location A, the Polaris will automatically adds the prefix like /root/namespace/A, related code https://github.com/apache/polaris/blob/17a359b43cf961a1261a7dfa7405ff47fb06b5f3/runtime/service/src/main/java/org/apache/polaris/service/catalog/iceberg/IcebergCatalog.java#L1244. I don’t think this is correct, it breaks the contract between Iceberg REST clients and servers. The REST Server should only modify the location when it’s missing.

The unit tests I added don't have the issue as it doesn't go through that route.

Reply:
In short, the location checking is fine. I will merge this PR, and file another to fix the view issue.